### DISCUSSION TITLE
Zoom in/out config or plugin

### DISCUSSION THREAD

**Discussion by rickywu** [ayamir/nvimdots#1352] [General]
Seems cannnot use <C-+> or <C--> to Zoom
any other plugin can do this or can config like neodie:

```
if vim.g.neovide == true then
  vim.api.nvim_set_keymap("n", "<C-=>", ":lua vim.g.neovide_scale_factor = math.min(vim.g.neovide_scale_factor + 0.1,  1.0)<CR>", { silent = true })
  vim.api.nvim_set_keymap("n", "<C-->", ":lua vim.g.neovide_scale_factor = math.max(vim.g.neovide_scale_factor - 0.1,  0.1)<CR>", { silent = true })
  vim.api.nvim_set_keymap("n", "<C-+>", ":lua vim.g.neovide_transparency = math.min(vim.g.neovide_transparency + 0.05, 1.0)<CR>", { silent = true })
  vim.api.nvim_set_keymap("n", "<C-_>", ":lua vim.g.neovide_transparency = math.max(vim.g.neovide_transparency - 0.05, 0.0)<CR>", { silent = true })
  vim.api.nvim_set_keymap("n", "<C-0>", ":lua vim.g.neovide_scale_factor = 0.5<CR>", { silent = true })
  vim.api.nvim_set_keymap("n", "<C-)>", ":lua vim.g.neovide_transparency = 0.9<CR>", { silent = true })
end
```

**Comment by Jint-lzxy** [comment]
You might want to use some Vimscript for this![^1] For example, you can change the first line to:[^2][^3]
```vim
let g:neovide_scale_factor = (g:neovide_scale_factor + 0.1) > 1.0 ? 1.0 : g:neovide_scale_factor + 0.1
```
If you add this to a keymap definition:
```lua
["n|<C-=>"] = map_cr("let g:neovide_scale_factor = (g:neovide_scale_factor + 0.1) > 1.0 ? 1.0 : g:neovide_scale_factor + 0.1")
	:with_noremap()
	:with_silent(),
```
it should work nicely! It's been running smoothly for me.

[^1]: I've often run into problems when tryna update Vim's built-in dictionaries in a keymap, but it turns out that Vimscript manages that much more effectively.
[^2]: Vim's `min()` and `max()` don't work with floats, so that's why we're using a ternary expression to avoid [E805](https://neovim.io/doc/user/eval.html#E805).
[^3]: If you're looking for better error handling in case `g:neovide_scale_factor` is undefined, you can use [`get()`](https://neovim.io/doc/user/builtin.html#get()-dict).
